File: Rotate String.py
# https://leetcode.com/problems/rotate-string/?envType=daily-question&envId=2024-11-03
# Rotate String

class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        
        for i in range(len(s)):

            if s == goal:
                return True

            s = s[1:] + s[0]

        return False

File: Rotate String.js
// https://leetcode.com/problems/rotate-string/?envType=daily-question&envId=2026-05-03
// Rotate String

/**
 * @param {string} s
 * @param {string} goal
 * @return {boolean}
 */
var rotateString = function(s, goal) {
    
    for(let i = 0; i < s.length ; i++) {
        if (s == goal) return true;

        s = s.substr(1, s.length) + s[0];
    }

    return false;

};